### PR TITLE
New Channel Guide

### PR DESCRIPTION
**Pull request by lancehalvorsen**
Currently this is a WIP with lots of work yet to do. I did want to bring some early visibility to this for anybody who is interested. // @chrismccord

I'm aiming to accomplish several things with this:
- break the explanation into small pieces that build logically on what has come before
- bring a sense of experimentation and verification to building the examples
- work as much as possible in iex to remove the need to build out a client

### CODE REVIEW

**Comment by lancehalvorsen** [comment]
I think this is good enough to merge at this point. @chrismccord if you could take a look for problems in this, I would really appreciate it. If it's ok, I'll move this over to phoenixframework.org tomorrow night.

closes #8

File: G_channels.md [hunk 1]
```diff
@@ -1,5 +1,385 @@
...
+
+- Channels
+
+Channels handle requests, so they are similar to Controllers, but there are two key differences. Channel requests can go both directions - incoming and outgoing. Channel connections also persist beyond a single request/response cycle. Channels are the highest level abstraction for realtime communication components in Phoenix.
+
+Each Channel will implement one or more clauses of each of these four callback functions - `join/3`, `leave/2`, `handle_in/3`, and `handle_out/3`.
+
+- PubSub
+
+The `Phoenix.PubSub` module contains functions which are the nuts and bolts of organizing Channel communication - creating topics, deleting topics, subscribing to topics, unsubscribing from topics, broadcasting messages on a topic, listing subscribers to a topic, checking if a topic is active, and checking if a topic exists.
```
Comment (chrismccord) [inline G_channels.md]: The `Phoenix.PubSub` module contains functions which are the nuts and bolts of organizing Channel communication - subscribing to topics, unsubscribing from topics, and broadcasting messages on a topic.

File: G_channels.md [hunk 2]
```diff
@@ -1,5 +1,385 @@
...
+
+It is worth noting that this module is intended for Phoenix's internal use. Channels use it under the hood to do much of their work. As end users, we shouldn't have any need to use PubSub directly.
+
```
Comment (chrismccord) [inline G_channels.md]: :+1:

**Comment by chrismccord** [comment]
@lancehalvorsen Nice work!!! I will make some small edits before we ship 0.9 since we've had some minor API changes. Thank you!

**Comment by lancehalvorsen** [comment]
@chrismccord thanks! I'll make that quick edit this evening. Thanks in advance for any 0.9 edits you can offer - will make for a smooth, coordinated release. :)

**Pull request merged**